modules/jwplayerVideoProvider.js

```diff
@@ -0,0 +1,997 @@
+import {
+  PROTOCOLS, API_FRAMEWORKS, VIDEO_MIME_TYPE, PLAYBACK_METHODS, PLACEMENT, VPAID_MIME_TYPE
+} from './videoModule/constants/ortb.js';
```

Comment:
The rule is still, "no cross-module imports". This is what gets built:

https://gist.github.com/dgirardi/3f9738dc3999ac3b4a433f299c10ccaf

All of these definitions are duplicated in every module that imports them. With the 'hook.js' pattern you adopted here you should have resolved the duplication of state, but code is still repeated. If we start adding more, distinct `videoProvider` modules that do the same imports, they would get repeated again.

It's a bit clunky, but I think the way to do what you want here is to remove `import`/`export`s and pass symbols you need through the `submodule` initializer - something like:

```
// videoModule/index.js
const sharedUtils = {...}
module('videoModule', function(submod) {
    registerSubmodule(submod(sharedUtils));
});
```

```
// someVideoProviderInstance.js
submodule('videoModule', function(sharedUtils) {
    return function submodFactory() {
            doStuffWith(sharedUtils);
    }
})